kotlin-inject-compiler-kapt/src/main/kotlin/me/tatarka/inject/compiler/kapt/ModelAst.kt

```diff
@@ -243,6 +244,18 @@ private class ModelAstClass(
             }.firstOrNull()
         }
 
+    override val constructors: List<AstConstructor>?
+        get() {
+            return ElementFilter.constructorsIn(element.enclosedElements).mapIndexed { index, constructor ->
+                ModelAstConstructor(
+                    this,
+                    this,
+                    constructor,
+                    kmClass?.constructors?.get(index)
```

Comment:
This assumption that the order of constructors is the same from both sources is notably shaky. I assume this problem is similar to what the TODO above refers to as well. My low familiarity with the used APIs hasn't allowed me to solve this properly yet. The tests seem to be passing though.

Reply:
Yep you are correct. I believe this can be done similar to how methods are matched. Look up the kotlin metadata, then use it's jvm signature to match the corresponding element. Would have to take a closer look to make sure.

Reply:
Thank you, will look into it.

Reply:
Updated with your recommendation.